src/scheduler/multi_threading.c

```diff
@@ -174,14 +174,14 @@ init_multi_threading(int nthreads)
 	pthread_mutex_init(&result_lock, &attr);
 	pthread_mutex_init(&general_lock, &attr);
 
-	if (nthreads < 1) {
+	num_cores = sysconf(_SC_NPROCESSORS_ONLN);
+	if (nthreads < 1 && num_cores > 2)
 		/* Create as many threads as half the number of cores */
-		num_cores = sysconf(_SC_NPROCESSORS_ONLN);
 		num_threads = num_cores / 2;
-	} else
+	else
 		num_threads = nthreads;
 
-	if (num_threads == 1)
+	if (num_threads <= 1)
```

Comment:
The 0 value is interpreted as "this is not set" elsewhere in the code, I can't set it to 1 by default because then there'd be no way to know if it was set to 1 explicitly by the admin.

Reply:
or did you mean set it to 1 here if it's 0 ? That actually is a good thought, I'll make that change.

Reply:
Yeah, I meant set it to 1 if it's less than one.